### PR TITLE
Move .write rule under .rule layer

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add .rule layer over .write rule:
  - `.write: <write rule>` -> `.rule: { .write: <write rule> }`
- Rename .write rule: `.write` -> `write`

Related issue: https://github.com/ainblockchain/ain-blockchain/issues/473

### CODE REVIEW

File: db/state-util.js [hunk 1]
```diff
@@ -169,9 +169,16 @@ function isValidJsObjectForStates(obj) {
...
  */
  function isValidRuleConfig(ruleConfigObj) {
-  if (!CommonUtil.isBool(ruleConfigObj) && !CommonUtil.isString(ruleConfigObj)) {
+  if (!CommonUtil.isDict(ruleConfigObj)) {
     return { isValid: false, invalidPath: CommonUtil.formatPath([]) };
   }
+  if (!ruleConfigObj.hasOwnProperty(RuleProperties.WRITE)) {
+    return { isValid: false, invalidPath: CommonUtil.formatPath([]) };
+  }
+  const writeConfigObj = ruleConfigObj[RuleProperties.WRITE];
```
Comment (liayoo) [inline db/state-util.js:178]: minor) how about just `writeConfig` since it's either a boolean or a string?

Comment (liayoo) [reply db/state-util.js:178]: Or writeProp, like ownersProp?

**Review by liayoo — APPROVED** [review]
LGTM with some minor comments. Thanks!

File: db/state-util.js (hunk 1 above)
Comment (platfowner) [reply db/state-util.js]: Just done. Thanks for the idea!

**Comment by platfowner** [comment]
Thanks for the fast review. Now being merged..

**Pull request merged**